pkg/block/block_test.go

```diff
@@ -218,3 +222,54 @@ func TestDelete(t *testing.T) {
 		testutil.Equals(t, 2, len(bkt.Objects()))
 	}
 }
+
+func TestMarkForDeletion(t *testing.T) {
+	defer leaktest.CheckTimeout(t, 10*time.Second)()
+
+	ctx := context.Background()
+
+	tmpDir, err := ioutil.TempDir("", "test-block-mark-for-delete")
+	testutil.Ok(t, err)
+	defer func() { testutil.Ok(t, os.RemoveAll(tmpDir)) }()
+
+	bkt := inmem.NewBucket()
+	{
+		blockWithoutDeletionMark, err := e2eutil.CreateBlock(ctx, tmpDir, []labels.Labels{
+			{{Name: "a", Value: "1"}},
+			{{Name: "a", Value: "2"}},
+			{{Name: "a", Value: "3"}},
+			{{Name: "a", Value: "4"}},
+			{{Name: "b", Value: "1"}},
+		}, 100, 0, 1000, labels.Labels{{Name: "ext1", Value: "val1"}}, 124)
+		testutil.Ok(t, err)
+		testutil.Ok(t, Upload(ctx, log.NewNopLogger(), bkt, path.Join(tmpDir, blockWithoutDeletionMark.String())))
+
+		testutil.Ok(t, MarkForDeletion(ctx, log.NewNopLogger(), bkt, blockWithoutDeletionMark))
+		exists, err := bkt.Exists(ctx, path.Join(blockWithoutDeletionMark.String(), metadata.DeletionMarkFilename))
+		testutil.Ok(t, err)
+		testutil.Equals(t, true, exists)
+	}
+	{
+		blockWithDeletionMark, err := e2eutil.CreateBlock(ctx, tmpDir, []labels.Labels{
+			{{Name: "a", Value: "1"}},
+			{{Name: "a", Value: "2"}},
+			{{Name: "a", Value: "3"}},
+			{{Name: "a", Value: "4"}},
+			{{Name: "b", Value: "1"}},
+		}, 100, 0, 1000, labels.Labels{{Name: "ext1", Value: "val1"}}, 124)
+		testutil.Ok(t, err)
+		testutil.Ok(t, Upload(ctx, log.NewNopLogger(), bkt, path.Join(tmpDir, blockWithDeletionMark.String())))
+
+		deletionMark, err := json.Marshal(metadata.DeletionMark{
+			ID:           blockWithDeletionMark,
+			DeletionTime: time.Now().Unix(),
+			Version:      metadata.DeletionMarkVersion1,
+		})
+		testutil.Ok(t, err)
+		testutil.Ok(t, bkt.Upload(ctx, path.Join(blockWithDeletionMark.String(), metadata.DeletionMarkFilename), bytes.NewReader(deletionMark)))
+
+		err = MarkForDeletion(ctx, log.NewNopLogger(), bkt, blockWithDeletionMark)
```

Comment:
we can inline those.

Reply:
On line 271, we are using err at two places, so we cannot inline it.
On line 263, we have two variables deletionMark and err, so we cannot inline it
On line 253, we have two variables blockWithDeletionMark and err. So it cannot be inlined as well

I am resolving this thread since we cannot inline.